docs/proposals/edgemesh-design.md

```diff
@@ -76,6 +76,149 @@ Add two endpoints definition:
 |ServiceBus|Target|Request Forward|
 |Router|Source|Resutst Forward, Service Discovery, Load Balance|
 
+<img src="../images/proposals/router-design.png">
+
+* R1: EdgeController watches on k8s apiserver for Services/Endpoints.
+* R2: EC get the service definition and create router rule definition and save in DB(sqlite).
+* R3: Router gets the rule definition from DB and initializes the listeners/handlers for source and target.
+* R4: Users/Applications create a request.
+* R5: Router receives the request and forwards it to the target based on rule.
+
+Since Router fetches rules from DB, in later versions it can be started as a different process than edge-controller. Also it enables router to retrieve all the rules after process restarts.
+### Router Low-level Design
+
+### Providers
+Providers are plugins to reach a service running on edge or on cloud. For example (servicebus,eventbus running in edge_core) 
+Providers can be classified into two types
+1) Source
+2) Target
+
+In Alpha1.0 only ServiceBus, REST and EventBus providers will be supported.
+
+Providers in router work in following way.
```

Comment:
I will appreciate a diagram to illustrate the following workflows.

Reply:
This workflow just explains the role of providers int the picture above. The diagram is the same :)

Reply:
It's confusing and inconsistent with the diagram above.